spt/OrangeBox/modules/vphysicsDLL.cpp

```diff
@@ -0,0 +1,83 @@
+#include "stdafx.h"
+
+#include "vphysicsDLL.hpp"
+
+#include "convar.h"
+
+#include "..\modules.hpp"
+#include "..\patterns.hpp"
+
+#define DEF_FUTURE(name) auto f##name = FindAsync(ORIG_##name, patterns::inputsystem::##name);
+#define GET_HOOKEDFUTURE(future_name) \
+	{ \
+		auto pattern = f##future_name.get(); \
+		if (ORIG_##future_name) \
+		{ \
+			DevMsg("[vphysics dll] Found " #future_name " at %p (using the %s pattern).\n", \
+			       ORIG_##future_name, \
+			       pattern->name()); \
+			patternContainer.AddHook(HOOKED_##future_name, (PVOID*)&ORIG_##future_name); \
+			for (int i = 0; true; ++i) \
+			{ \
+				if (patterns::vphysics::##future_name.at(i).name() == pattern->name()) \
+				{ \
+					patternContainer.AddIndex((PVOID*)&ORIG_##future_name, i, pattern->name()); \
+					break; \
+				} \
+			} \
+		} \
+		else \
+		{ \
+			DevWarning("[vphysics dll] Could not find " #future_name ".\n"); \
+		} \
+	}
+
+void VPhysicsDLL::Hook(const std::wstring& moduleName,
+                       void* moduleHandle,
+                       void* moduleBase,
+                       size_t moduleLength,
+                       bool needToIntercept)
+{
+	Clear();
+	m_Name = moduleName;
+	m_Base = moduleBase;
+	m_Length = moduleLength;
+	patternContainer.Init(moduleName);
+
+	// use patterns of hooked functions in clientDLL to figure out which version we're on and therefore which offset to use
+	int isgOff = 0;
+#if defined(SSDK2007)
+	if (clientDLL.ORIG_UTIL_TraceRay)
+	{
+		std::string vers = clientDLL.patternContainer.FindPatternName((PVOID*)&clientDLL.ORIG_UTIL_TraceRay);
```

Comment:
Kinda weird deciding vphysics.dll offsets based on client.dll, for example it means it might break for mods unnecessarily. But I'll leave the decision to @Jukspa.